packages/drivers/odsp-driver/src/compactSnapshotWriter.ts

```diff
@@ -0,0 +1,140 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { stringToBuffer } from "@fluidframework/common-utils";
+import { IBlob, ISnapshotTree } from "@fluidframework/protocol-definitions";
+import { snapshotMinReadVersion } from "./compactSnapshotParser";
+import { ISequencedDeltaOpMessage } from "./contracts";
+import { NodeCore, ReadBuffer, TreeBuilder } from "./zipItDataRepresentationUtils";
+
+/**
+ * Writes header section of the snapshot.
+ * @param node - tree node to serialize to
+ * @param snapshotSeqNumber - seq number at which snapshot is created.
+*/
+function writeHeaderSection(node: NodeCore, snapshotSeqNumber: number) {
+    node.addString("MinReadVersion");
```

Comment:
For readability, consider creating function addProperty(node, propertyName, propertyValue) and calling it here.

Reply:
But why always have extra code which will go through some kind of switch statement whenever we need to add a property. Also we would be adding extra code. Imo this seems readable enough?

Reply:
```typescript
const addProperty(node, a, b) => { node.addString(a); node.addString(b); }

function writeHeaderSection(node: NodeCore, snapshotSeqNumber: number) {
   addProperty(node, "MinReadVersion", snapshotMinReadVersion);
   addProperty(node, "CreateVersion", snapshotMinReadVersion);
   addProperty(node, "SnapshotSequenceNumber", snapshotSeqNumber);
}
```

Happy to bet that it will be not bigger in minimized version :)
That's because addProperty would become a single letter, but public names (like addString) stay as is.
So 4 "addString" are removed, bug small function is added.